src/main/java/org/folio/service/pieces/PieceStorageService.java

```diff
@@ -187,9 +190,19 @@ private Future<List<Piece>> getPieceChunkByLineIds(Collection<String> poLineIds,
       .map(PieceCollection::getPieces);
   }
 
-  private static boolean shouldFilterPiecesForTenant(String centralTenantId, RequestContext requestContext) {
+  static String getQueryForUserTenants(List<String> userTenants, String query) {
+    if (CollectionUtils.isEmpty(userTenants)) {
+      return query;
+    }
+    var cql = convertIdsToCqlQuery(userTenants, "receivingTenantId");
+    var matchNullTenantExpr = getCqlExpressionForFieldNullValue("receivingTenantId");
+    cql = combineCqlExpressions("or", cql, matchNullTenantExpr);
+    return combineCqlExpressions("and", query, cql);
+  }
```

Comment:
this part is a little bit hard to follow. Suggest:
```
static String getQueryForUserTenants(List<String> userTenants, String query) {
    if (CollectionUtils.isEmpty(userTenants)) {
        return query;
    }

    String tenantCql = convertIdsToCqlQuery(userTenants, "receivingTenantId");
    String nullTenantCql = getCqlExpressionForFieldNullValue("receivingTenantId");
    String combinedTenantCql = combineCqlExpressions("or", tenantCql, nullTenantCql);

    return combineCqlExpressions("and", query, combinedTenantCql);
}
```

Reply:
Both comments addressed